ibis/backends/duckdb/__init__.py

```diff
@@ -494,6 +494,49 @@ def read_csv(
             con.exec_driver_sql(view)
         return self.table(table_name)
 
+    def read_geo(
+        self,
+        source: str,
+        table_name: str | None = None,
+        **kwargs: Any,
+    ) -> ir.Table:
+        """Register a GEO file as a table in the current database.
+
+        Parameters
+        ----------
+        source
```

Comment:
I just happen to try reading directly from a url, more specifically I tried and things work, I don't have a way from reading from s3 or the other cloud options, but given the nature of `st_read` my guess is this should work for any cloud format,
```python 
In [1]: import ibis
In [2]: con = ibis.duckdb.connect()
In [3]: t = con.read_geo("https://raw.githubusercontent.com/ibis-project/testing-data/master/geojson/zones.geojson")
In [4]: t.geom.area().to_pandas()
Out[5]: 
0      7.903953e+07
1      1.439095e+08
2      3.168508e+07
3      8.023733e+06
4      5.041488e+07
           ...     
258    3.974125e+07
259    4.263797e+07
260    3.468847e+06
261    6.957826e+06
262    6.636394e+06
Name: GeoArea(geom), Length: 263, dtype: float64
```

That being said there seems to be a regression that will be fixed  https://github.com/duckdb/duckdb_spatial/issues/182